src/bindings/c/include/openvino/ov_model.h

```diff
@@ -0,0 +1,215 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for ov_model C API
+ *
+ * @file ov_model.h
+ */
+
+#pragma once
+
+#include "ov_common.h"
+
+// Model
+/**
+ * @defgroup Model Model
+ * @ingroup openvino_c
+ * Set of functions representing of Model and Node.
+ * @{
+ */
+
+/**
+ * @brief Release the memory allocated by ov_model_t.
+ * @ingroup Model
+ * @param model A pointer to the ov_model_t to free memory.
+ */
+OPENVINO_C_API(void) ov_model_free(ov_model_t* model);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup Model
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup Model
+ * @param model A pointer to the ov_model_t.
+ * @param input_nodes A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_nodes);
+
+/**
+ * @brief Get the tensor name of ov_output_node list by index.
+ * @ingroup Model
+ * @param nodes A pointer to the ov_output_node_list_t.
+ * @param idx Index of the input tensor
+ * @param tensor_name A pointer to the tensor name.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_node_get_any_name_by_index(ov_output_node_list_t* nodes, size_t idx, char** tensor_name);
+
+/**
+ * @brief Get the tensor name of node.
+ * @ingroup Model
+ * @param node A pointer to the ov_output_const_node_t.
+ * @param tensor_name A pointer to the tensor name.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_node_get_any_name(ov_output_const_node_t* node, char** tensor_name);
```

Comment:
It is same with the C++ exposed object to caller, do we need split more and bring user  more difficulty to use it?

Reply:
It will be better to have separate headers for different headers (as we have in C++), because it allows more clear describe structures in the file.
For example the end user do not expected that functions and structures what describes node are defined in `model.h` file.

Reply:
ok, will split them.

Reply:
done